vault/dynamodb.tf

```diff
@@ -0,0 +1,85 @@
+locals {
+  dynamodb_table_name = "vault-${var.environment}-${var.project}"
```

Comment:
I'm not convinced about this. IMO this should have been the right approach from the beginning, as previously the dynamodb table name was hardcoded to "vault-dynamodb-backend", which would prevent multiple Vault setups in the same AWS account.
On the other hand, changing this now to a dynamic name introduces a huge overhead when rolling this out to the current Vault setups, as we'll need to migrate data to another db, which will require some downtime and it'll probably be time consuming. So I'm not sure if it's worth the effort.